shapeworks_cloud/core/rest.py

```diff
@@ -50,6 +53,31 @@ def _create(self, request, **kwargs):
 
         return Response({**{'id': instance.id}, **serializer.data}, status=201)
 
+    @swagger_auto_schema(
+        manual_parameters=[
+            openapi.Parameter(
+                'filename',
+                openapi.IN_QUERY,
+                description='Search for a filename',
+                type=openapi.TYPE_STRING,
+            )
+        ],
+        request_body=no_body,
+        responses={
+            200: 'Search result list',
+        },
+    )
+    @action(methods=['GET'], detail=False)
+    def search(self, request, **kwargs):
```

Comment:
Since there's already a listing endpoint, would it be clearer to use a django filter e.g. https://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend?

Reply:
that is a much better idea

Reply:
Turns out filter backends work with querysets. Because `name` is a computed property and not a proper field, we need to filter a list of models instead. It would be nice to get the automatic query parameter, but I don't see a way to do it with a filter backend.

Reply:
Yeah. I think it could work using https://django-filter.readthedocs.io/en/stable/ref/filters.html#method, but it would be clunky. My main reason for wanting it is just the separation of filtering code but oh well.

We should keep in mind that dynamically computing the name is causing some friction, in the event that it happens again.